kotlinx-coroutines-core/common/src/channels/Broadcast.kt

```diff
@@ -119,6 +125,13 @@ private open class BroadcastCoroutine<E>(
         val processed = _channel.close(cause)
         if (!processed && !handled) handleCoroutineException(context, cause)
     }
+
+    // The BroadcastChannel could be also closed
+    override fun close(cause: Throwable?): Boolean {
```

Comment:
This change is not enough. 
Now if you'll close `BroadcastChannel` with exception different from `CancellationException`, it will be reported to global exception handler, because `onCancelled` will try to close _channel second time